src/Listener/JsonApiListener.php

```diff
@@ -0,0 +1,281 @@
+            'Crud.setFlash' => ['callable' => [$this, 'setFlash'], 'priority' => 5],
+            'Crud.beforeRender' => ['callable' => [$this, 'respond'], 'priority' => 100],
+            'Crud.beforeRedirect' => ['callable' => [$this, 'respond'], 'priority' => 100]
+        ];
+    }
+
+    /**
+     * beforeHandle
+     *
+     * Called before the crud action is executed.
+     *
+     * @param \Cake\Event\Event $event Event
+     * @return void
+     */
+    public function beforeHandle(Event $event)
+    {
+        $this->_checkRequestMethods();
+        $this->_decodeIncomingJsonApiData();
+    }
+
+    /**
+     * Selects an specific Crud view class to render the output
+     *
+     * @param \Crud\Event\Subject $subject Subject
+     * @return \Cake\Network\Response
+     */
+    public function render(Subject $subject)
+    {
+        $controller = $this->_controller();
+        $controller->viewBuilder()->className('Crud.JsonApi');
+
+        // Validate configuration before creating corresponding viewVars
+        if ($this->config('urlPrefix') !== null && !is_string($this->config('urlPrefix'))) {
+            throw new CrudException('JsonApiListener configuration option `urlPrefix` only accepts a string');
+        }
+        $controller->set([
+            '_urlPrefix' => $this->config('urlPrefix')
+        ]);
+
+        if ($this->config('withJsonApiVersion')) {
+            $controller->set([
+                '_withJsonApiVersion' => $this->config('withJsonApiVersion')
+            ]);
+        }
+
+        if ($this->config('meta')) {
+            if (!is_array($this->config('meta'))) {
+                throw new CrudException('JsonApiListener configuration option `meta` only accepts an array');
+            }
+            $controller->set([
+                '_meta' => $this->config('meta')
+            ]);
+        }
+
+        if (!is_array($this->config('include'))) {
+            throw new CrudException('JsonApiListener configuration option `include` only accepts an array');
+        }
+        $controller->set([
+            '_include' => $this->config('include')
+        ]);
+
+        if (!is_array($this->config('fieldSets'))) {
+            throw new CrudException('JsonApiListener configuration option `fieldSets` only accepts an array');
+        }
+        $controller->set([
+            '_fieldSets' => $this->config('fieldSets')
+        ]);
+
+        // Set required viewVar with ORM AssociationCollection for the current
+        // model so it can be used to generate the `relationships` nodes in the
+        // response.
+        //
+        // Please note that we are removing associated models not found in
+        // the find() result to prevent `null` relationships appearing in
+        // the response.
+        $controllerName = $controller->name; // e.g. Countries
+        $entityName = Inflector::singularize($controllerName); // e.g. Country
+        $table = $controller->$controllerName; // table object
```

Comment:
Correct, changed as suggested in https://github.com/FriendsOfCake/crud/pull/445/commits/6a70b299a063ead55c934455f5fb491c811426bb